src/lib/http-server.ts

```diff
@@ -15,67 +15,74 @@
 
 /**
  * HTTP Server library
+ *
  * @module lib/http-server
  */
 
-'use strict';
-
-const http = require('http');
-
-const server = Symbol('server');
+import { AssertionError } from 'assert';
+import { Server, RequestListener, createServer } from 'http';
+import { AddressInfo } from 'net';
 
 /**
  * Provides a restartable wrapper for http.CreateServer().
  */
-class HttpServer {
-  /**
-   * @callback requestHandler
-   * @param {http.IncomingMessage} request The incoming message.
-   * @param {http.ServerResponse} response The server response.
-   */
+export class HttpServer {
+  // Can't use symbol as an indexer because of https://github.com/microsoft/TypeScript/issues/1863
+  // Switching to ECMAScript private fields https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/#ecmascript-private-fields
+  #server: Server;
```

Comment:
Can't use symbol as an indexer because of https://github.com/microsoft/TypeScript/issues/1863

Switching to ECMAScript private fields https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/#ecmascript-private-fields